internal/unionstore/memdb_arena.go

```diff
@@ -331,6 +337,47 @@ func (l *memdbVlog) getValue(addr memdbArenaAddr) []byte {
 	return block[valueOff:lenOff:lenOff]
 }
 
+// Get value, and possibly write a new vlog entry in the latest stage to let the locking phase be aware of it via InspectStage.
+// It should be used where the read result affect users or upper level modules. Temporary flags need to be unset in such cases.
+func (l *memdbVlog) getValue(addr memdbArenaAddr) (value []byte) {
+	if !l.memdb.enableTemporaryFlags {
+		return l.pureGetValue(addr)
+	}
+	hdrOffset := addr.off - memdbVlogHdrSize
```

Comment:
Could the flag check be executed here so if it's false the function returns directly.

Reply:
Sorry I don't get your point. Seems to me it just does what you say?